Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -329,6 +330,49 @@ extension BrowserViewController: URLBarDelegate {
         hideSearchController()
         updateInContentHomePanel(tabManager.selectedTab?.url)
     }
+
+    func urlBar(urlBar: URLBarView, suggestionForText text: String, complete: String? -> Void) {
+        let options = QueryOptions(filter: text, filterType: FilterType.Url, sort: QuerySort.Frecency)
+        profile.history.get(options, complete: { (cursor: Cursor) in
```

Comment:
Thanks for the feedback. Yeah, I have a TODO comment about reusing the last suggestion if it still fits [here](https://github.com/mozilla/firefox-ios/pull/300/files#diff-ff4fcceb6ce0c9bf38fb315a4959ffacR65) (assuming that's what you're talking about). It probably wouldn't be much extra work to add that now, so I can do that.